forge-game/src/main/java/forge/game/ability/effects/SacrificeAllEffect.java

```diff
@@ -84,8 +84,7 @@ public void resolve(SpellAbility sa) {
 
         Map<Integer, Card> cachedMap = Maps.newHashMap();
         Map<AbilityKey, Object> params = AbilityKey.newMap();
-        CardZoneTable table = new CardZoneTable(game.copyLastStateBattlefield(), CardCollection.EMPTY);
-        AbilityKey.addCardZoneTableParams(params, table);
+        CardZoneTable zoneMovements = AbilityKey.addCardZoneTableParams(params, sa);
 
         for (Card sac : list) {
             final Card lKICopy = CardUtil.getLKICopy(sac, cachedMap);
```

Comment:
@Hanmac 
You added the `cachedMap` in some places, but I don't see the point here:
these are different cards so LKI wouldn't get reused anyway?

Reply:
just for the corner case when you sacrifice multiple cards at once
that may be attached to each other

the cachedMap is just for when getting an LKI copy, to not copy something that already got copied.

otherwise we might look into this using LKI that already got copied in LastStateBattlefield?

Reply:
i didn't mean it like that

more like:
`final Card lKICopy = zoneMovements.getLastStateBattlefield().get(sac);`

this way, we can use the LKI that was already generated for LastStateBattlefield?

Reply:
that's how it will work?
`getLKICopy` checks the prefilled cache first
this way there's some safety if for some reason the LastState doesn't have it and just creates a new one instead

Reply:
i mean it shouldn't even be possible

you can only Sacrifice from the Battlefield, and at that moment, LastState should already have the card

so there is no need to make an extra CacheMap for this, right?

Reply:
In an ideal world yes?
Seems stable without, will run some longer AI tests as backup